src/app/clusters/application-basic-server/application-basic-delegate.h

```diff
@@ -28,22 +30,66 @@ namespace app {
 namespace Clusters {
 namespace ApplicationBasic {
 
+using ApplicationBasicApplicationType = chip::app::Clusters::ApplicationBasic::Structs::ApplicationBasicApplication::Type;
+
+class DLL_EXPORT CatalogVendorApp
+{
+public:
+    CatalogVendorApp(){};
+    CatalogVendorApp(CatalogVendorApp * app)
+    {
+        catalogVendorId = app->catalogVendorId;
+        Platform::CopyString(applicationId, sizeof(applicationId), app->applicationId);
+    };
+    CatalogVendorApp(uint16_t vendorId, const char * appId) { Set(vendorId, appId); };
+
+    bool Matches(CatalogVendorApp app)
```

Comment:
Here, and in other places: should probably pass by const ref.  Followup is fine.